lib/heap_sort.rb

```diff
@@ -1,8 +1,48 @@
-
+require_relative 'min_heap'
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity:  O(nlogn)
+# Space Complexity: O(1)
+def heapsort(list)
```

Comment:
Outstanding work!  You got the in-place O(1) space complexity Heapsort.  Well done!